### ISSUE TITLE
Question about ValueDice's nonlinear loss

### ISSUE THREAD

**Issue by gunshi** [google-research/google-research#331]
Hi, 
This is a question for the repo for the ValueDice paper, (perhaps @peterjliu or Ilya Kostrikov is the relevant person?) 
I see that there is a softmax (instead of a log-sum-exp) involved to compute the J_linear part of the valuedice loss : https://github.com/google-research/google-research/blob/7c7fdac0d57ef262896776785a59d824026a076b/value_dice/value_dice.py#L27 
was wondering if you could explain why that is?
Is this something to do with wanting to normalise this part of the loss for some reason?
Would really appreciate any clarification.

Thanks!
Gunshi

**Comment by peterjliu** [comment]
I have nothing to do with this

**Comment by gunshi** [comment]
@ikostrikov maybe?

**Comment by ikostrikov** [comment]
@gunshi see equation 12 in https://arxiv.org/pdf/1801.04062.pdf

In this case the softmax part corresponds to analytically computed gradients of log sum exp.

**Comment by gunshi** [comment]
Hey @ikostrikov, can I also ask for help with the derivation of, or even the intuition behind the replay buffer regularisation version of the objective? It looks to me like in eqn 18 of the arxiv version, we're maximising and minimising the same quantity in the log and linear parts of the expression (for the replay buffer samples).
Thanks!

**Comment by ikostrikov** [comment]
@gunshi I think it might be easier to explain for the toy MDP that we used in the paper. When the regularization is not used, the values are not defined for certain states and actions. Also it's similar to soft targets used in Improved Techniques for Training GANs.

**Comment by gunshi** [comment]
I see! so is it fair to say that we would actually want the policy roll-outs to be somewhat sub-optimal? Because really good policy roll-outs (that almost look like the expert) are probably sending a contradictory learning signal to the critic, yes?
Thanks for your responses, they're really helpful!

**Comment by ikostrikov** [comment]
Yes, it's better to have rather suboptimal rollouts with a larger coverage for the replay buffer.

**Comment by gunshi** [comment]
got it, thanks!
